src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -347,6 +361,22 @@ void VfsCfApi::onHydrationJobFinished(HydrationJob *job)
     }
 }
 
+void VfsCfApi::onHydrationJobCanceled(HydrationJob *job)
+{
+    const auto folderPath = job->localPath();
+    const auto folderRelativePath = job->folderPath();
+
+    // Remove placeholder file because there might be already pumped
+    // some data into it
+    QFile::remove(folderPath + folderRelativePath);
+
+    // Create a new placeholder file
+    SyncJournalFileRecord record;
+    params().journal->getFileRecord(folderRelativePath, &record);
+    const auto item = SyncFileItem::fromSyncJournalFileRecord(record);
```

Comment:
**If** that's **really** required (see other comment above), then at least it should check the db read actually succeeded and the record is valid, otherwise you could have a malformed item here.

Reply:
@er-vin Okay but how should we recover if the item is malformed? Or should we check if the record is valid before the removal of the placeholder and if the record is invalid then just return early from the function?

Reply:
Yes, probably best to do the db operation first. And leave the file in place if that fails. At least that'd avoid unwanted deletion on next sync.

Reply:
@er-vin https://github.com/nextcloud/desktop/pull/3028